### PR TITLE
fixes 'dict object' has no attribute 'stdout' and wrong yum updates

### LINKED ISSUE #3: update the yum package index
Hi,

yum * latest **does not** update the package index, but updates every installed package on the system.
```
- name: update the yum package index
  yum:
    name: "*"
    state: latest
  when: ansible_pkg_mgr == "yum"
```

This needs to be removed, as a system update should absolutely not be included in a steamcmd install. If you really are concerned about yum cache check the option "update_cache".

-> https://docs.ansible.com/ansible/latest/modules/yum_module.html

cheers
Michael

### LINKED ISSUE #5: steam_login gets overwritten
Task `linking steamclient.so to the ~/.steam/sdk32/steamclient.so directory` fails, because var `steam_login` gets overwritten, when task is skipped.
More info:
https://github.com/ansible/ansible/issues/17500

### PR DESCRIPTION
**Pull request by michaelsstuff**
fixes #3 
fixes #5

### CODE REVIEW

File: tasks/packages.yml [hunk 1]
```diff
@@ -5,12 +5,6 @@
     update_cache: yes
   when: ansible_pkg_mgr == "apt"
 
-- name: update the yum package index
```
Comment (lutangar) [inline tasks/packages.yml:8]: I'd rather use `update_cache` option as mentioned in #3 instead of removing this.

File: tasks/requirements.yml [hunk 2]
```diff
@@ -1,9 +0,0 @@
-- name: check if anonymous steam account has no password
```
Comment (lutangar) [inline tasks/requirements.yml]: Why remove this in this PR? I'll be happy to discuss this in another PR or issue if you consider this useless.

File: tasks/setup.yml [hunk 3]
```diff
@@ -12,40 +12,69 @@
...
+  when: stat_steamcmd_result.stat.exists == False
 
 - name: extract the contents to cmd directory
   unarchive:
     src: "{{ steamcmd_directory }}steamcmd_linux.tar.gz"
     dest: "{{ steamcmd_directory }}"
     copy: no
+  when: stat_steamcmd_result.stat.exists == False
+
+- name: attempt to log into steam anonymously
```
Comment (lutangar) [inline tasks/setup.yml:33]: Nice catch :+1: 
Whereas I understand the workaround, I would prefer something like the following based on `block` should work:
https://github.com/ansible/ansible/issues/4297#issuecomment-356427588
For what I can recall, there should be a way to tackle this with a conditionnal include of another set of tasks.

**Review by lutangar — CHANGES_REQUESTED** [review]

File: tasks/requirements.yml (hunk 2 above)
Comment (michaelsstuff) [reply tasks/requirements.yml]: Because tasks/requirements.yml is never called, or did I miss something?

File: tasks/packages.yml (hunk 1 above)
Comment (michaelsstuff) [reply tasks/packages.yml:8]: ok, but then we would need to make two tasks, one for apt, one for yum. `update_cache` only works together when state is `present `or `latest` and the generic package module does not have a update_cache option
Did it in michaelsstuff/ansible-role-steamcmd@5e10d8d

File: tasks/setup.yml (hunk 3 above)
Comment (michaelsstuff) [reply tasks/setup.yml:33]: block also does not work. Even if the block is conditional and skipped, the register inside still gets the skipped info. see https://github.com/ansible/ansible/issues/4297#issuecomment-386962997
I think the set_fact works fine here, but if you have another solution I will test it.



```
- block:
    - name: attempt to log into steam anonymously
      command: "{{ steamcmd_directory }}steamcmd.sh +quit"
      async: "{{ steamcmd_login_timeout}}"
      poll: 5
      register: output
      no_log: yes
  when: steamcmd_steam.username == "anonymous"

- debug:
    var: output
    verbosity: 2
```

Result:

```
TASK [ansible-role-steamcmd : attempt to log into steam anonymously] ************************************************************************************************************************************************************************
task path: /root/.ansible/roles/ansible-role-steamcmd/tasks/setup.yml:34
skipping: [localhost] => {"censored": "the output has been hidden due to the fact that 'no_log: true' was specified for this result", "changed": false}

TASK [ansible-role-steamcmd : debug] ********************************************************************************************************************************************************************************************************
task path: /root/.ansible/roles/ansible-role-steamcmd/tasks/setup.yml:42
ok: [localhost] => {
    "output": {
        "changed": false,
        "skip_reason": "Conditional result was False",
        "skipped": true
    }
}


```

File: tasks/requirements.yml (hunk 2 above)
Comment (michaelsstuff) [reply tasks/requirements.yml]: readded in cfc768e312d2da16c1ca7103f8f1236d56c57a08

**Comment by lutangar** [comment]
Thanks a lot @michaelsstuff one more thing though, before this PR is merged, could you remove commits 1926fbe0663da63444146e8492c2494461ab5bda and cfc768e312d2da16c1ca7103f8f1236d56c57a08 from the history?
I like to keep git history as relevant as possible.

**Comment by michaelsstuff** [comment]
Done @lutangar

**Comment by lutangar** [comment]
Thanks a lot :tada:

**Pull request merged**